### ISSUE TITLE
Cannot add a component to a `FluentCalendar` using `withComponent` after calling `getFluentTarget`.

### ISSUE THREAD

**Issue by hacetin** [ical4j/ical4j#758]
**Describe the bug**
Cannot add a component to a `FluentCalendar` using `withComponent` after calling `getFluentTarget`.

`withComponent` gets `CalendarComponent` parameter `default FluentCalendar withComponent(CalendarComponent component)`

`getFluentTarget` returns `Component`: `public Component getFluentTarget()`

So, it throws a `type mismatch` exception.

**To Reproduce**
I use it in Scala, so I will share a Scala example but it should happen in Java the same way.

```scala
val organizer = new property.Organizer("mailto:email@email.com")
val event = new component.VEvent(
      Instant.now(),
      Instant.now(),
      "Event name",
    )
  .withProperty(organizer)
  .getFluentTarget
val calender = new Calendar()
  .withProperty(property.immutable.ImmutableVersion.VERSION_2_0)
  .withProperty(property.immutable.ImmutableCalScale.GREGORIAN)
  .withProperty(property.immutable.ImmutableMethod.CANCEL)
  .getFluentTarget

calender
  .withProperty(property.immutable.ImmutableStatus.VEVENT_CANCELLED)
  .withComponent(event)
  .toString

```
 `.withComponent(meeting)` will throw type mismatch exception.

![image](https://github.com/user-attachments/assets/883d9b8c-ee23-4a91-a0c3-92ba447a58f1)

**Expected behavior**
`getFluentTarget` should return and object with proper type

**Environment (please complete the following information):**
 - MacOS
 - Java Version 17
 - iCal4j Version 4.0.6

**Comment by hacetin** [comment]
probably happenned after this change: https://github.com/ical4j/ical4j/commit/95ea373204270af152cb124b6b63de8a246221a4#diff-5d94906202461f19df4ec21e0f924d9a8ed0ecdc050697bf1a129194940b1dcdL196

**Comment by benfortuna** [comment]
Yeah, actually I think it's a result of the design, whereby `FluentCalendar` is more like a builder interface used for creating (but not modifying) objects.

To modify a calendar after building you can try using PropertyContainer/ComponentContainer methods (e.g. `add()`, `with()`, etc.), which work in a similar way to the fluent interfaces.

**Comment by DanielNovak** [comment]
I think this will also work, it's a bit hacky but internally it should be the same type:

```
calender
  .withProperty(property.immutable.ImmutableStatus.VEVENT_CANCELLED)
  .withComponent(event.fluentTarget as VEvent)
  .toString
```
(in Kotlin)
